File: PART2-OOP/advertisment/src/BaseAdvertising.java
public class BaseAdvertising {
    private int id;
    private int clicks;
    private int views;

    public BaseAdvertising() {
    }

    public BaseAdvertising(int id) {
        this.id = id;
    }

    public int getClicks() {
        System.out.println("Base--getClicks: " + clicks);
        return clicks;
    }

    public int getViews() {
        System.out.println("Base--getViews: " + views);
        return views;
    }

    public void incClicks() {
        this.clicks += 1;
    }

    public void incViews() {
        this.views += 1;
    }

    public void describeMe(){
        //TODO: Return String
    }
}

File: PART3-PYTHON/advertisement/base_advertising.py

class BaseAdvertising:
    """--- BaseAdvertising Class Description ---
    Ad and Advertiser inherit this class.
    This class contains common fields and methods between these two classes.\n"""

    def __init__(self, base_id=None):
        self.base_id = base_id
        self.__clicks = 0
        self.__views = 0

    def get_clicks(self):
        return self.__clicks

    def get_view(self):
        return self.__views

    def inc_views(self):
        self.__views += 1

    def inc_clicks(self):
        self.__clicks += 1

    def describe_me(self):
        return self.__doc__


